aquifer-git.js

```diff
@@ -100,185 +98,96 @@ module.exports = (Aquifer, AquiferGitConfig) => {
       return;
     }
 
-    // If we have a name without an email, or an email with no name (like XOR),
-    // then we cannot create a custom signature and need to bail out.
-    if (!options.name !== !options.email) {
-      callback('Both name and email options are required for a custom commit signature.');
-      return;
-    }
-
     // Create the destination directory and initiate the promise chain.
     mktemp.createDir('aquifer-git-XXXXXXX')
+    // Clone the repository.
+    .then((destPath_) => {
+      Aquifer.console.log('Cloning the repository into ' + destPath_ + '...', 'status');
+      destPath = destPath_;
+      return run.invoke('git clone ' + options.remote + ' ' + destPath);
+    })
+
+    // Prepare the repository for the build.
+    .then(() => {
+      Aquifer.console.log('Checking out branch: ' + options.branch + '...', 'status');
+      return run.invoke('bash -c "cd ' + path.join(Aquifer.projectDir, destPath) + ' && git checkout ' + options.branch + '"');
```

Comment:
@mattgrill you're right. It's not difficult. I'm not opposed to that change to the run API, but it should probably be handled outside this PR (and on the aquifer project first) so we can focus on removing the dependency on nodegit in this PR.